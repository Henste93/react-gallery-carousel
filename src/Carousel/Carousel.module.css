.imagesWrapper {
  width: 100%;
  height: 100%;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  touch-action: pan-x pinch-zoom;
}

.images {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
  transition-duration: 0s;
  transition-property: transform;
  transition-timing-function: ease-out;
  touch-action: pinch-zoom;
}

.images:focus {
  outline: none;
}
