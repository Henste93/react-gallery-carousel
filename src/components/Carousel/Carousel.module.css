.carouselWrapper {
  position: relative;
  user-select: none;
  display: flex;
  background-color: #fff;
  flex-flow: column;
  align-items: stretch;
  box-sizing: border-box;
}

.maxCarouselWrapper {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 2147483647; /* int max */
  width: 100%;
  height: 100%;
  /*noinspection CssInvalidPropertyValue*/
  /*max-width: stretch;*/
  /*noinspection CssInvalidPropertyValue*/
  /*max-height: stretch;*/
  overflow: hidden;
  background-color: #fff;
  transform: scale(1);
  transform-origin: top center;
  display: flex;
  flex-flow: column;
  align-items: stretch;
  box-sizing: border-box;
}

.carousel {
  width: 100%;
  height: 100%;
  overflow: hidden;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  touch-action: auto;
  box-sizing: border-box;
  flex: 0 1 auto;
}

.gallery {
  background-color: #000;
}

[data-is-not-keyboard-user='true'] *:focus {
  outline: none;
}
