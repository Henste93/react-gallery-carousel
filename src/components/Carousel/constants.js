// constant(s)

// Swiping down longer than MAX_SWIPE_DOWN_DISTANCE will not have more visual changes
export const MAX_SWIPE_DOWN_DISTANCE = 1500; // px

export const WIDGET_POSITIONS = [
  'topLeft',
  'topCenter',
  'topRight',
  'centerLeft',
  'centerCenter',
  'centerRight',
  'bottomLeft',
  'bottomCenter',
  'bottomRight',
  'top',
  'bottom'
];
// place large widgets those have multiple focusable elements at the end,
// so that keyboard users can get to their desired elements easier and faster.

export const WIDGET_POSITIONS_WITH_RTL = [
  'topRight',
  'topCenter',
  'topLeft',
  'centerRight',
  'centerCenter',
  'centerLeft',
  'bottomRight',
  'bottomCenter',
  'bottomLeft',
  'top',
  'bottom'
];
