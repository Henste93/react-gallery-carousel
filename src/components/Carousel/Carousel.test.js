import React from 'react';
import renderer from 'react-test-renderer';
import Carousel from '../Carousel';
const { describe, it, expect } = global;

const imageIDs = Array(6)
  .fill(1)
  .map((_, i) => i + 1);
const images = imageIDs.map((imageID) => {
  return {
    src: `https://placedog.net/8${imageID}0/6${imageID}0?id=${imageID}`,
    alt: `Dog No. ${imageID}. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated.`,
    thumbnail: `https://placedog.net/8${imageID}/6${imageID}?id=${imageID}`
  };
});

// snapshot testing
describe('Carousel', () => {
  it('is truthy', () => {
    expect(Carousel).toBeTruthy();
  });

  global.IntersectionObserver = function () {
    return {
      observe: () => {},
      disconnect: () => {}
    };
  };

  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: (query) => ({
      matches: false,
      media: query,
      onchange: null,
      addEventListener: () => {},
      removeEventListener: () => {}
    })
  });

  it('renders the same as snapshot', () => {
    const component = renderer.create(<Carousel images={images} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('can be maximized and renders the same as snapshot', () => {
    const component2 = renderer.create(
      <Carousel images={images} isMaximized />
    );
    const tree2 = component2.toJSON();
    expect(tree2).toMatchSnapshot();
  });
});
