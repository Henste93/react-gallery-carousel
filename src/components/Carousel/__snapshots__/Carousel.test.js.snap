// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel can be maximized and renders the same as snapshot 1`] = `
Array [
  <div
    className="carousel gallery"
  />,
  <div
    className="maxCarousel gallery"
  />,
  <div
    className="maxCarousel gallery"
    data-is-not-keyboard-user="true"
    style={Object {}}
  >
    <div
      className="slidesWrapper"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUpCapture={[Function]}
      tabIndex={0}
    >
      <ul
        className="slides"
      >
        <li
          className="slide"
        >
          <figure
            aria-label="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="figure"
          >
            <img
              alt="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="image"
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
            <img
              alt="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="thumbnail"
              loading="lazy"
              onError={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
          </figure>
        </li>
        <li
          className="slide"
        >
          <figure
            aria-label="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="figure"
          >
            <img
              alt="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="image"
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
            <img
              alt="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="thumbnail"
              loading="lazy"
              onError={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
          </figure>
        </li>
        <li
          className="slide"
        >
          <figure
            aria-label="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="figure"
          >
            <img
              alt="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="image"
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
            <img
              alt="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="thumbnail"
              loading="lazy"
              onError={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
          </figure>
        </li>
        <li
          className="slide"
        >
          <figure
            aria-label="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="figure"
          >
            <img
              alt="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="image"
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
            <img
              alt="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="thumbnail"
              loading="lazy"
              onError={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
          </figure>
        </li>
        <li
          className="slide"
        >
          <figure
            aria-label="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="figure"
          >
            <img
              alt="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="image"
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
            <img
              alt="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="thumbnail"
              loading="lazy"
              onError={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
          </figure>
        </li>
        <li
          className="slide"
        >
          <figure
            aria-label="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="figure"
          >
            <img
              alt="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="image"
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
            <img
              alt="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="thumbnail"
              loading="lazy"
              onError={[Function]}
              src=""
              style={
                Object {
                  "objectFit": "contain",
                }
              }
            />
          </figure>
        </li>
      </ul>
      <div
        className="widgetWrapper topLeft"
      >
        <button
          aria-disabled={false}
          aria-label="Start Autoplay"
          className="button "
          onClick={[Function]}
        >
          <div
            className="iconWrapper square"
          >
            <svg
              aria-labelledby="play-title play-desc"
              height="30"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 30 30"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="play-title"
              >
                Play
              </title>
              <desc
                id="play-desc"
              >
                Equilateral triangle pointing right
              </desc>
              <path
                d="m15 22.5-15 7.5v-15-15l15 7.5 15 7.5z"
                fill="#fff"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        className="widgetWrapper textWrapper topCenter"
        tabIndex="-1"
      >
        <span
          className="text"
        >
          1
           / 
          6
        </span>
      </div>
      <div
        className="widgetWrapper topRight"
      >
        <button
          aria-disabled={false}
          aria-label="Minimize Slides"
          className="button "
          onClick={[Function]}
        >
          <div
            className="iconWrapper square"
          >
            <svg
              aria-labelledby="min-title min-desc"
              height="30"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 30 30"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="min-title"
              >
                Minimize
              </title>
              <desc
                id="min-desc"
              >
                4 small arrows pointing inwards
              </desc>
              <g
                fill="#fff"
              >
                <path
                  d="m0 5h10v5h-10z"
                />
                <path
                  d="m0 20h10v5h-10z"
                />
                <path
                  d="m10 0v10h-5v-10z"
                />
                <path
                  d="m10 20v10h-5v-10z"
                />
                <path
                  d="m20 5h10v5h-10z"
                />
                <path
                  d="m20 20h10v5h-10z"
                />
                <path
                  d="m25 0v10h-5v-10z"
                />
                <path
                  d="m25 20v10h-5v-10z"
                />
              </g>
            </svg>
          </div>
        </button>
      </div>
      <div
        className="widgetWrapper centerLeft"
      >
        <button
          aria-disabled={false}
          aria-label="Go to Previous Slide"
          className="button "
          onClick={[Function]}
        >
          <div
            className="iconWrapper rectangle"
          >
            <svg
              aria-labelledby="left-title left-desc"
              height="40"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 20 40"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="left-title"
              >
                Left
              </title>
              <desc
                id="left-desc"
              >
                Obtuse triangle pointing left
              </desc>
              <path
                d="m20 0v40l-20-20z"
                fill="#fff"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        className="widgetWrapper centerRight"
      >
        <button
          aria-disabled={false}
          aria-label="Go to Next Slide"
          className="button "
          onClick={[Function]}
        >
          <div
            className="iconWrapper rectangle"
          >
            <svg
              aria-labelledby="right-title right-desc"
              height="40"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 20 40"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="right-title"
              >
                Right
              </title>
              <desc
                id="right-desc"
              >
                Obtuse triangle pointing right
              </desc>
              <path
                d="m0 40v-40l20 20z"
                fill="#fff"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div
      className="thumbnailsWrapper"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUpCapture={[Function]}
      onWheel={[Function]}
    >
      <ul
        className="thumbnails"
      >
        <li
          className="thumbnail currentThumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          tabIndex={0}
        >
          <img
            alt="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            src=""
          />
        </li>
        <li
          className="thumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          tabIndex={0}
        >
          <img
            alt="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            src=""
          />
        </li>
        <li
          className="thumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          tabIndex={0}
        >
          <img
            alt="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            src=""
          />
        </li>
        <li
          className="thumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          tabIndex={0}
        >
          <img
            alt="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            src=""
          />
        </li>
        <li
          className="thumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          tabIndex={0}
        >
          <img
            alt="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            src=""
          />
        </li>
        <li
          className="thumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          tabIndex={0}
        >
          <img
            alt="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            src=""
          />
        </li>
      </ul>
    </div>
  </div>,
]
`;

exports[`Carousel renders the same as snapshot 1`] = `
<div
  className="carousel gallery"
  data-is-not-keyboard-user="true"
>
  <div
    className="slidesWrapper"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUpCapture={[Function]}
    tabIndex={0}
  >
    <ul
      className="slides"
    >
      <li
        className="slide"
      >
        <figure
          aria-label="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="figure"
        >
          <img
            alt="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
          <img
            alt="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="thumbnail"
            loading="lazy"
            onError={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
        </figure>
      </li>
      <li
        className="slide"
      >
        <figure
          aria-label="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="figure"
        >
          <img
            alt="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
          <img
            alt="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="thumbnail"
            loading="lazy"
            onError={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
        </figure>
      </li>
      <li
        className="slide"
      >
        <figure
          aria-label="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="figure"
        >
          <img
            alt="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
          <img
            alt="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="thumbnail"
            loading="lazy"
            onError={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
        </figure>
      </li>
      <li
        className="slide"
      >
        <figure
          aria-label="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="figure"
        >
          <img
            alt="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
          <img
            alt="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="thumbnail"
            loading="lazy"
            onError={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
        </figure>
      </li>
      <li
        className="slide"
      >
        <figure
          aria-label="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="figure"
        >
          <img
            alt="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
          <img
            alt="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="thumbnail"
            loading="lazy"
            onError={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
        </figure>
      </li>
      <li
        className="slide"
      >
        <figure
          aria-label="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="figure"
        >
          <img
            alt="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
          <img
            alt="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="thumbnail"
            loading="lazy"
            onError={[Function]}
            src=""
            style={
              Object {
                "objectFit": null,
              }
            }
          />
        </figure>
      </li>
    </ul>
    <div
      className="widgetWrapper topLeft"
    >
      <button
        aria-disabled={false}
        aria-label="Start Autoplay"
        className="button "
        onClick={[Function]}
      >
        <div
          className="iconWrapper square"
        >
          <svg
            aria-labelledby="play-title play-desc"
            height="30"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 30 30"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="play-title"
            >
              Play
            </title>
            <desc
              id="play-desc"
            >
              Equilateral triangle pointing right
            </desc>
            <path
              d="m15 22.5-15 7.5v-15-15l15 7.5 15 7.5z"
              fill="#fff"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      className="widgetWrapper textWrapper topCenter"
      tabIndex="-1"
    >
      <span
        className="text"
      >
        1
         / 
        6
      </span>
    </div>
    <div
      className="widgetWrapper topRight"
    >
      <button
        aria-disabled={false}
        aria-label="Maximize Slides"
        className="button "
        onClick={[Function]}
      >
        <div
          className="iconWrapper square"
        >
          <svg
            aria-labelledby="max-title max-desc"
            height="30"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 30 30"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="max-title"
            >
              Maximize
            </title>
            <desc
              id="max-desc"
            >
              4 small arrows pointing outwards
            </desc>
            <g
              fill="#fff"
            >
              <path
                d="m0 0h10v5h-10z"
              />
              <path
                d="m0 25h10v5h-10z"
              />
              <path
                d="m5 0v10h-5v-10z"
              />
              <path
                d="m5 20v10h-5v-10z"
              />
              <path
                d="m20 0h10v5h-10z"
              />
              <path
                d="m20 25h10v5h-10z"
              />
              <path
                d="m30 0v10h-5v-10z"
              />
              <path
                d="m30 20v10h-5v-10z"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
    <div
      className="widgetWrapper centerLeft"
    >
      <button
        aria-disabled={false}
        aria-label="Go to Previous Slide"
        className="button "
        onClick={[Function]}
      >
        <div
          className="iconWrapper rectangle"
        >
          <svg
            aria-labelledby="left-title left-desc"
            height="40"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 20 40"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="left-title"
            >
              Left
            </title>
            <desc
              id="left-desc"
            >
              Obtuse triangle pointing left
            </desc>
            <path
              d="m20 0v40l-20-20z"
              fill="#fff"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      className="widgetWrapper centerRight"
    >
      <button
        aria-disabled={false}
        aria-label="Go to Next Slide"
        className="button "
        onClick={[Function]}
      >
        <div
          className="iconWrapper rectangle"
        >
          <svg
            aria-labelledby="right-title right-desc"
            height="40"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 20 40"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="right-title"
            >
              Right
            </title>
            <desc
              id="right-desc"
            >
              Obtuse triangle pointing right
            </desc>
            <path
              d="m0 40v-40l20 20z"
              fill="#fff"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div
    className="thumbnailsWrapper"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUpCapture={[Function]}
    onWheel={[Function]}
  >
    <ul
      className="thumbnails"
    >
      <li
        className="thumbnail currentThumbnail"
        onMouseUpCapture={[Function]}
        role="button"
        tabIndex={0}
      >
        <img
          alt="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          aria-label="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="image"
          loading="lazy"
          onError={[Function]}
          src=""
        />
      </li>
      <li
        className="thumbnail"
        onMouseUpCapture={[Function]}
        role="button"
        tabIndex={0}
      >
        <img
          alt="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          aria-label="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="image"
          loading="lazy"
          onError={[Function]}
          src=""
        />
      </li>
      <li
        className="thumbnail"
        onMouseUpCapture={[Function]}
        role="button"
        tabIndex={0}
      >
        <img
          alt="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          aria-label="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="image"
          loading="lazy"
          onError={[Function]}
          src=""
        />
      </li>
      <li
        className="thumbnail"
        onMouseUpCapture={[Function]}
        role="button"
        tabIndex={0}
      >
        <img
          alt="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          aria-label="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="image"
          loading="lazy"
          onError={[Function]}
          src=""
        />
      </li>
      <li
        className="thumbnail"
        onMouseUpCapture={[Function]}
        role="button"
        tabIndex={0}
      >
        <img
          alt="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          aria-label="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="image"
          loading="lazy"
          onError={[Function]}
          src=""
        />
      </li>
      <li
        className="thumbnail"
        onMouseUpCapture={[Function]}
        role="button"
        tabIndex={0}
      >
        <img
          alt="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          aria-label="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
          className="image"
          loading="lazy"
          onError={[Function]}
          src=""
        />
      </li>
    </ul>
  </div>
</div>
`;
